mkosi/__init__.py

```diff
@@ -3585,26 +3607,37 @@ def make_minimal_btrfs(args: CommandLineArguments, root: str, for_cache: bool) -
         )
         f.truncate(args.root_size)
 
-        command = ["mkfs.btrfs", "-L", "root", "-d", "single", "-m", "single", "--shrink", "--rootdir", root, f.name]
-        try:
-            run(command)
-        except subprocess.CalledProcessError:
-            # The --shrink option was added in btrfs-tools 4.14.1, before that it was the default behaviour.
-            # If the above fails, let's see if things work if we drop it
-            command.remove("--shrink")
+        command = ["mkfs.btrfs", "-L", label, "-d", "single", "-m", "single", "--rootdir", root, f.name]
+
+        if args.minimal:
+            command.append("--shrink")
+            try:
+                run(command)
+            except subprocess.CalledProcessError:
+                # The --shrink option was added in btrfs-tools 4.14.1, before that it was the default behaviour.
+                # If the above fails, let's see if things work if we drop it
+                command.remove("--shrink")
+                run(command)
```

Comment:
Instead of adding and then removing it again, this might be clearer
```suggestion
            try:
                run(command + ["--shrink"])
            except subprocess.CalledProcessError:
                # The --shrink option was added in btrfs-tools 4.14.1, before that it was the default behaviour.
                # If the above fails, let's see if things work without it
                run(command)
```